### PR TITLE
/.github/workflows/ci-check-repo.yaml: Iterating on github actions check repo

### PR DESCRIPTION
**Pull request by coffeegoddd**
looking into fixing this: https://github.com/liquidata-inc/dolt/pull/767/checks?check_run_id=808971285#step:4:296
```
2020/06/25 21:04:02 Error running `git merge-base remotes/origin/db/ci-github-actions-check-repo remotes/origin/master` to find merge parent: exit status 128
exit status 1
```

the `exit status 128` is from `//go/utils/checkcommiters/main.go` ... im thinking could be the go version? i had to use `1.13` here to get past the error that occurred when i used go version `^1.13` regarding the `-mod=readonly` flag....

### CODE REVIEW

File: .github/workflows/ci-check-repo.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+jobs:
+  test:
+    name: Check Repo
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Go 1.13
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.13
```
Comment (coffeegoddd) [inline .github/workflows/ci-check-repo.yaml:15]: must be `1.13` as `^1.13` resulted in a flag error for `go get -mod=readonly ./...`

File: .github/workflows/ci-check-repo.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        working-directory: ./go
+        # Keep this in sync with //go/utils/prepr/prepr.sh.
+        run: |
+          go get -mod=readonly ./...
+          ./utils/repofmt/check_fmt.sh
+          ./Godeps/verify.sh
+          ./utils/checkcommitters/check_pr.sh
+          go vet -mod=readonly ./...
+          go run -mod=readonly ./utils/copyrightshdrs/
+          go test -mod=readonly -test.v ./...
```
Comment (coffeegoddd) [inline .github/workflows/ci-check-repo.yaml:30]: Is it necessary to run the tests here as well as in https://github.com/liquidata-inc/dolt/pull/762 ?

Comment (reltuk) [reply .github/workflows/ci-check-repo.yaml:30]: No, we don't need to do this here.

File: .github/workflows/ci-check-repo.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.13
+        id: go
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Check all
+        working-directory: ./go
+        # Keep this in sync with //go/utils/prepr/prepr.sh.
```
Comment (reltuk) [inline .github/workflows/ci-check-repo.yaml:22]: Can we update `prepr.sh` to say "Keep this in sync with `//.github/workflows/ci-check-repo.yaml`?

File: .github/workflows/ci-check-repo.yaml (hunk 1 above)
Comment (reltuk) [reply .github/workflows/ci-check-repo.yaml:15]: Can we upgrade the script to work with go > `1.13`? Maybe it's fine to just drop this statement in general. Or do `GOFLAGS="-mod readonly" go build ./...` to check everything builds without modifications to `go.mod`?

**Pull request merged**